src/main/java/Duke.java

```diff
@@ -1,10 +1,189 @@
+            break;
+        case Deadline.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Deadline(taskName, date));
+            }
+            break;
+        case Event.TASK_TYPE:
+            details = getTaskNameAndDate(detailFragments, taskType);
+            if (details != null) {
+                taskName = details[0];
+                date = details[1];
+                records.add(new Event(taskName, date));
+            }
+            break;
+        }
+        if (details != null || taskType.equals(Todo.TASK_TYPE)) {
+            int numberOfRecords = records.size();
+            System.out.println("Got it. I've added this task:");
+            System.out.println("\t" + records.get(numberOfRecords - 1));
+            System.out.printf("Now you have %d tasks in the list.\n", numberOfRecords);
+        }
+    }
+
+    private static String[] getTaskNameAndDate(String[] detailFragments, String taskType) {
+        int slashElementIndex;
+        boolean hasSlashElement = false;
+        boolean isTypeMatch = false;
+        for (slashElementIndex = 0; slashElementIndex < detailFragments.length - 1; slashElementIndex++) {
+            if (detailFragments[slashElementIndex].charAt(0) == '/') {
+                hasSlashElement = true;
+                break;
+            }
+        }
+        if (!hasSlashElement) {
+            System.out.println("Invalid input! (No date/time provided)");
+            return null;
+        }
+
+        switch (taskType) {
+        case Deadline.TASK_TYPE:
+            if (detailFragments[slashElementIndex].substring(1).equals("by")) {
+                isTypeMatch = true;
+            }
+            break;
+        case Event.TASK_TYPE:
+            if (detailFragments[slashElementIndex].substring(1).equals("at")) {
+                isTypeMatch = true;
+            }
+            break;
+        }
+        if (!isTypeMatch) {
+            System.out.println("Invalid input! (keywords not matching)");
+            return null;
+        }
+
+        StringBuilder taskName = new StringBuilder();
+        StringBuilder dueDay = new StringBuilder();
+        for (int i = 0; i < slashElementIndex; i++) {
+            taskName.append(detailFragments[i]);
+        }
+        for (int i = slashElementIndex + 1; i < detailFragments.length; i++) {
+            dueDay.append(detailFragments[i]);
+        }
+        return new String[]{taskName.toString(), dueDay.toString()};
+    }
+
+    private static void showList() {
+        System.out.println("Here is your task List:");
+        for (int i = 0; i < records.size(); i++) {
+            System.out.println((i + 1) + ". " + records.get(i));
+        }
+    }
+
+    private static void quitProgram() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+}
```

Comment:
Spacings are very consistent and the method names are simple yet understandable.

Reply:
Thanks!